src/compiler/checker.ts

```diff
@@ -24909,11 +25038,19 @@ namespace ts {
                 }
                 else {
                     const isStatic = hasModifier(member, ModifierFlags.Static);
-                    const names = isStatic ? staticNames : instanceNames;
-
+                    let names;
                     const name = member.name;
+                    if (!name) {
+                        return;
+                    }
+                    if (isPrivateIdentifier(name)) {
+                        names = privateIdentifiers;
+                    }
+                    else {
+                        names = isStatic ? staticNames : instanceNames;

```

Comment:
Updated `checkClassForDuplicateDeclarations` to catch this sort of case:

```ts
class A {
    static #foo;
    #foo  // error
}
```

Related new test cases are:
- privateNamesUnique-4.ts
- privateNamesInNestedClasses-2.ts